### DISCUSSION TITLE
Help Getting the Current Index

### DISCUSSION THREAD

**Discussion by kerns** [dromru/react-photoswipe-gallery#1293] [General]
In my next.js based implementation, I am having a lot of trouble pushing a new location on slide change – so this is a Hail Mary with the hope someone stumbles over this and can rescue me. 🙏 ...the issue seems to be pswpInstance.currItem is never defined.

```
const Home: NextPage = ({ images }: { images: ImageProps[] }) => {
  const router = useRouter();

...

const onBeforeOpen = (pswpInstance) => {
    pswpInstance.on("change", () => {
      // const currentSlide = pswpInstance.currItem;
      // const currentSlideId = images[currentSlide.index].id;
      console.log("slide was changed");
      // router.push(`/?photoId=${currentSlideId}`, `/photo/${currentSlideId}`, {
      //   shallow: true,
      // });
    });
  };

...

```

**Comment by kerns** [comment]
(and it's not bc it is commented out – sorry, that is only because all of the commented out code breaks things. "slide was changed" is being logged when the other things that throw errors are removed)

**Comment by akhmadullin** [comment]
Hi

`Photoswipe` instance has no `currItem` property, that's why `pswpInstance.currItem`  is always equals to `undefined`.

If you need to get object with data about current slide, you should use `pswpInstance.currSlide`.
